### PR TITLE
Raise error if user slices NDCube with Ellipsis.

### LINKED ISSUE #216: Slicing with Ellipsis gives wrong axis types
### Description

Slicing an `NDCube` object with an Ellipsis leads to incorrect slicing of the `world_axis_physical_types`. For example, from the documentation:

```python
import numpy as np
import astropy.wcs
from ndcube import NDCube

data = np.ones((3, 4, 5))
wcs_input_dict = {
'CTYPE1': 'WAVE    ', 'CUNIT1': 'Angstrom', 'CDELT1': 0.2, 'CRPIX1': 0, 'CRVAL1': 10, 'NAXIS1': 5,
'CTYPE2': 'HPLT-TAN', 'CUNIT2': 'deg', 'CDELT2': 0.5, 'CRPIX2': 2, 'CRVAL2': 0.5, 'NAXIS2': 4,
'CTYPE3': 'HPLN-TAN', 'CUNIT3': 'deg', 'CDELT3': 0.4, 'CRPIX3': 2, 'CRVAL3': 1, 'NAXIS3': 3}
input_wcs = astropy.wcs.WCS(wcs_input_dict)

my_cube = NDCube(data, input_wcs)
```

Doing `my_cube[:, :, 0]` will give the axis types `('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat')`, as expected:

```
In [4]: my_cube[:, :, 0]
Out[4]:
NDCube
---------------------
WCS Keywords

Number of WCS axes: 3
CTYPE : 'WAVE'  'HPLT-TAN'  'HPLN-TAN'
CRVAL : 1e-09  0.5  1.0
CRPIX : 0.0  2.0  2.0
PC1_1 PC1_2 PC1_3  : 1.0  0.0  0.0
PC2_1 PC2_2 PC2_3  : 0.0  1.0  0.0
PC3_1 PC3_2 PC3_3  : 0.0  0.0  1.0
CDELT : 2.0000000000000002e-11  0.5  0.4
NAXIS : 1  4  3
---------------------
Length of NDCube: [3. 4.] pix
Axis Types of NDCube: ('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat')
```

Slicing as  `my_cube[..., 0]` should give the same result. The data seem correctly sliced, but it will keep the second axis type as wavelength, which is incorrect:

```
In [5]: my_cube[..., 0]
Out[5]:
NDCube
---------------------
WCS Keywords

Number of WCS axes: 3
CTYPE : 'WAVE'  'HPLT-TAN'  'HPLN-TAN'
CRVAL : 1e-09  0.5  1.0
CRPIX : 0.0  2.0  2.0
PC1_1 PC1_2 PC1_3  : 1.0  0.0  0.0
PC2_1 PC2_2 PC2_3  : 0.0  1.0  0.0
PC3_1 PC3_2 PC3_3  : 0.0  0.0  1.0
CDELT : 2.0000000000000002e-11  0.5  0.4
NAXIS : 5  4  1
---------------------
Length of NDCube: [3. 4.] pix
Axis Types of NDCube: ('custom:pos.helioprojective.lon', 'em.wl')
```

### System Details

  - ndcube Version: 1.2.1
  - SunPy  Version: 0.9.8
  - Python Version: 3.7.3
  - OS information: macOS 10.14.6, python from conda

### PR DESCRIPTION
**Pull request by DanRyanIrish**
### Description

ndcube 1.x does not support slicing ```NDCube```s with ```Ellipses```.  However, when a user tries this, the operation does not fail, but rather slices the data correctly, but the FITS-WCS incorrectly.  This PR causes the operation to fail with an informative error.

Fixes #216

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
Getting an ```InvocationError``` and ```InterpreterError```.  Presumably this is a problem with the testing setup, not this PR itself?  @Cadair?

**Comment by Cadair** [comment]
yeah the fails are the same as #223 see #225 we should fix them in a new PR.

File: ndcube/utils/wcs.py [hunk 1]
```diff
@@ -183,6 +183,10 @@ def _wcs_slicer(wcs, missing_axes, item):
         new_wcs = wcs.slice(item_checked)
     # if it a tuple like [0:2, 0:3, 2] or [0:2, 1:3]
     elif isinstance(item, tuple):
+        # Ellipsis slicing is currently not supported.
+        # Raise an error if user tries to slice by ellipsis.
+        if type(Ellipsis) in [type(item_i) for item_i in item]:
```
Comment (Cadair) [inline ndcube/utils/wcs.py]: Can't you do this with:
```suggestion
        if Ellipsis in item:
```
?

**Comment by Cadair** [comment]
You should probably add a regression test :grinning:

**Comment by DanRyanIrish** [comment]
A regression test?!?!?!

**Comment by Cadair** [comment]
yeah so we don't break this later :stuck_out_tongue:

**Comment by DanRyanIrish** [comment]
@Cadair I don't understand the difference between a regression test and just running all the existing tests...

**Comment by DanRyanIrish** [comment]
Apart from a test, is there anything blocking you from approving this PR?

**Pull request merged**